File: multiStructure/core/api-clients/rest-client-templates/ts/apiClient.ts
export class ApiClient {
  private baseUrl: string;
  private token: string | null = null;

  constructor(baseUrl: string) {
    this.baseUrl = baseUrl;
  }

  async login(username: string, password: string): Promise<string> {
    const response = await fetch(`${this.baseUrl}/auth/login`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ username, password }),
    });

    const data = await response.json();
    this.token = data.token;
    return this.token;
  }

  private async request(path: string, options: RequestInit = {}): Promise<Response> {
    const headers: Record<string, string> = {
      'Content-Type': 'application/json',
      ...(this.token ? { Authorization: `Bearer ${this.token}` } : {}),
      ...(options.headers as Record<string, string> || {}),
    };

    return fetch(`${this.baseUrl}${path}`, {
      ...options,
      headers,
    });
  }

  async get(path: string): Promise<any> {
    const response = await this.request(path);
    return response.json();
  }

  async post(path: string, body: any): Promise<any> {
    const response = await this.request(path, {
      method: 'POST',
      body: JSON.stringify(body),
    });
    return response.json();
  }

  async put(path: string, body: any): Promise<any> {
    const response = await this.request(path, {
      method: 'PUT',
      body: JSON.stringify(body),
    });
    return response.json();
  }

  async delete(path: string): Promise<any> {
    const response = await this.request(path, { method: 'DELETE' });
    return response.json();
  }
}

File: multiStructure/core/api-clients/rest-client-templates/csharp/ApiClient.cs
using System.Net.Http.Json;

namespace Core.ApiClients;

public class ApiClient
{
    private readonly HttpClient _httpClient;
    private string? _token;

    public ApiClient(string baseUrl)
    {
        _httpClient = new HttpClient
        {
            BaseAddress = new Uri(baseUrl),
            DefaultRequestHeaders = { { "Accept", "application/json" } }
        };
    }

    public async Task<string> LoginAsync(string username, string password)
    {
        var response = await _httpClient.PostAsJsonAsync("auth/login", new { username, password });
        var data = await response.Content.ReadFromJsonAsync<LoginResponse>();
        _token = data?.Token;
        return _token ?? string.Empty;
    }

    private void SetAuthHeader()
    {
        if (!string.IsNullOrEmpty(_token))
        {
            _httpClient.DefaultRequestHeaders.Authorization =
                new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", _token);
        }
    }

    public async Task<T?> GetAsync<T>(string path)
    {
        SetAuthHeader();
        return await _httpClient.GetFromJsonAsync<T>(path);
    }

    public async Task<T?> PostAsync<T>(string path, object body)
    {
        SetAuthHeader();
        var response = await _httpClient.PostAsJsonAsync(path, body);
        return await response.Content.ReadFromJsonAsync<T>();
    }

    public async Task<T?> PutAsync<T>(string path, object body)
    {
        SetAuthHeader();
        var response = await _httpClient.PutAsJsonAsync(path, body);
        return await response.Content.ReadFromJsonAsync<T>();
    }

    public async Task DeleteAsync(string path)
    {
        SetAuthHeader();
        await _httpClient.DeleteAsync(path);
    }
}

public class LoginResponse
{
    public string? Token { get; set; }
    public int ExpiresIn { get; set; }
}
